labs-ios-starter/Surveys/Model Controllers/TopicController.swift

```diff
@@ -70,36 +69,20 @@ class TopicController {
         networkService.loadData(using: request) { result in
             switch result {
             case let .success(data):
-                guard let topics = self.networkService.decode(to: [Topic].self,
-                                                              data: data,
-                                                              moc: CoreDataManager.shared.mainContext) else {
+                guard self.networkService.decode(to: [Topic].self,
+                                                 data: data,
+                                                 moc: CoreDataManager.shared.mainContext) != nil else {
                     print("Error decoding topics")
                     completion(.failure(.badDecode))
                     return
                 }
-                var userTopics: [Topic] = []
-                if !all {
-                    guard let user = ProfileController.shared.authenticatedUserProfile else {
-                        print("🛑! User isn't logged in!")
-                        completion(.failure(.unauthorized))
-                        return
-                    }
-                    // This is going to be quite the expensive check if there are a lot of topics with a lot of members
-                    for topic in topics {
-                        if topic.leaderId == user.id { // member is leader
-                            userTopics.append(topic)
-                        } else if let members = topic.members { // member is user
-                            if members.contains(user) {
-                                userTopics.append(topic)
-                            }
-                        }
-                    }
-                }
-                completion(.success(userTopics))
+
+                try? CoreDataManager.shared.saveContext()
+
+                completion(.success(nil))
```

Comment:
I see why you changed to optional topics now, you can use the Void type in its place, and do

```suggestion
                completion(.success(Void()))
```